lib/rdf/prefix_map.ex

```diff
@@ -7,12 +7,16 @@ defmodule RDF.PrefixMap do
 
   alias RDF.IRI
 
-  @type prefix    :: atom | String.t
-  @type namespace :: atom | String.t | IRI.t
+  @type prefix    :: atom
+  @type namespace :: IRI.t
 
-  @type prefix_map :: %{prefix => namespace}
+  @type coercible_prefix    :: atom | String.t
+  @type coercible_namespace :: atom | String.t | IRI.t
 
-  @type conflict_resolver :: (prefix, namespace, namespace -> namespace)
+  @type prefix_map :: %{coercible_prefix => coercible_namespace}
```

Comment:
In this type only the strict types should be allowed

```suggestion
  @type prefix_map :: %{prefix => namespace}
```

Reply:
Thanks, fixed.